internal/worker/client.go

```diff
@@ -138,11 +149,97 @@ func NewClientUnix(conf ClientConfig) *Client {
 			},
 		},
 	}
-
-	return &Client{
+	client := &Client{
 		server:    server,
 		requester: requester,
 	}
+	err = client.register()
+	if err != nil {
+		panic(err)
+	}
+	go client.workerHeartbeat()
+	return client
+}
+
+func (c *Client) register() error {
```

Comment:
Nitpick: this seems to be registering a worker only, so maybe we could call this `registerWorker`?